cupy/core/_routines_statistics.pyx

```diff
@@ -59,7 +59,10 @@ cdef ndarray _ndarray_mean(ndarray self, axis, dtype, out, keepdims):
         result = cub.cub_reduction(self, cub.CUPY_CUB_SUM, axis, dtype, out,
                                    keepdims)
         if result is not None:
-            result /= (self.size / result.size)
+            if result.dtype.kind in ['f', 'c']:
+                result /= (self.size / result.size)
+            else:
+                result = result / (self.size / result.size)
```

Comment:
I think if `out` is given, the first branch should be taken.

(More precisely, `/=` does not work for integer arrays. I think we need another branch `if out is not None and result.dtype.kind in ('i', 'u')`  to use `//=`.)

Reply:
Looking into [how NumPy handles it](https://github.com/numpy/numpy/blob/c4d5e12879c19c2655b5620fa1c18242b14d5141/numpy/core/_methods.py#L149-L154), there is basically just one case, but it uses `casting='unsafe'` to handle the issue.

I think this would be the equivalent here:
```Python
navg = self.size // result.size
result = cupy.true_divide(result, navg , out=result, casting='unsafe')
```

Reply:
That sounds better 👍
(I don't think you have to reassign `result` though)